subprojects/functional/src/main/java/org/gradle/internal/Either.java

```diff
@@ -34,9 +37,54 @@
         return new Right<>(value);
     }
 
-    public abstract <U> U fold(Function<L, U> l, Function<R, U> r);
+    /**
+     * Is this a left?
+     */
+    public abstract boolean isLeft();
 
-    public abstract <U> Either<L, U> map(Function<R, U> r);
+    /**
+     * Is this a right?
+     */
+    public abstract boolean isRight();
+
+    /**
+     * Take the value if this is a left.
+     */
+    public abstract Optional<L> ifLeft();
```

Comment:
💄 I'd call this `getLeft()` so in Kotlin and Groovy code would read `either.left.ifPresent { ... }` instead of `either.ifLeft().ifPresent { ... }`.